PyInstaller/depend/analysis.py

```diff
@@ -100,6 +105,43 @@ def __init__(self, pyi_homepath, user_hook_dirs=None, *args, **kwargs):
             os.path.join(self._homepath, 'PyInstaller', 'loader', 'rthooks.dat')
         )
 
+    def msg(self, *args, **kwargs):
+        self._msg(*args, **kwargs)
+
+    def _msg(self, lvl, s, *args):
+        """
+        Print a debug message with the given level
+        """
+        sinfo = None
+        level = None
+
+        # Note: these levels are completely arbitrary and may be adjusted if needed.
+        if s:
+            msg = "%s %s" % (s, ' '.join(map(repr, args)))
+            if lvl <= 1:
+                level = WARNING
+            elif lvl <= 2:
+                level = INFO
+            elif lvl <= 4:
```

Comment:
What about using a mapping here:
```
LEVEL_MAPPING = {0: WARNING, 1:WARNING, 2: INFO, 3: DEBUG}
…
level = LEVEL.MAPPING.get(level)
```
This will make any other value (esp `4`) into `None` and negative values should not occur anyway. Or do they?

You could also save globally importing WARNING, INFO and DEBUG.(see line 38).